charm/docs/tutorial/getting-started.md

```diff
@@ -0,0 +1,122 @@
+# Getting Started
+
+## What you’ll do
+- Deploy the NetBox charm.
+- Use an external Redis server
+- Integrate with the PostgreSQL K8s charm.
+- Integrate with S3 for storage.
+- Expose the NetBox charm with Traefik k8s.
+- Create a super user.
+
+Through the process, you'll verify the workload state, and log in to
+your NetBox instance.
+
+## Requirements
+- Juju 3 installed.
+- Juju controller that can create a model of type kubernetes.
+- Read/write access to a S3 compatible server with a bucket created.
+- Configuration compatible with the traefik-k8s charms. In the case of MicroK8S this can be achieved with the metallb addon.
+
+For more information about how to install Juju, see [Get started with Juju](https://juju.is/docs/olm/get-started-with-juju).
+
+
+## Setting up a Tutorial Model
+
+To manage resources effectively and to separate this tutorial's workload from
+your usual work, we recommend creating a new model using the following command.
+
+```
+juju add-model netbox-tutorial
+```
+
+## Deploy the NetBox charm
+
+Deploy the NetBox charm, with all its mandatory requirements (PostgreSQL, Redis and S3).
+
+### Deploy the charms:
+```
+juju deploy netbox
+```
+
+At this point NetBox should be blocked as there is no S3 integration for
+storage, Redis or PostgreSQL.
+
+Set the allowed hosts. In this example every host is allowed. For a production environment
+only the used hosts should be allowed.
+```
+juju config nebox django_allowed_hosts='*'
+```
+
+### Redis
+
+NetBox requires Redis to work. Currently you can configure Redis with 
+configuration options (this will be changed to an integration in the 
+next version).
+
+Deploy redis-k8s:
+```
+juju deploy redis-k8s --channel=latest/edge
+```
+
+Get the admin_password:
+```
+juju run redis-k8s/0 get-initial-admin-password
+```
+
+Configure NetBox using the previous admin password:
+```
+juju config netbox redis_hostname=redis-k8s-0.redis-k8s-endpoints redis_password=<admin_password>
+```
+
+### Deploy PostgreSQL
+
+NetBox requires PostgreSQL to work. Deploy and integrate with:
+```
+juju deploy postgresql-k8s --channel 14/stable --trust
+juju integrate postgresql-k8s netbox
+```
+
+### Deploy s3-integrator
+
+NetBox requires a S3 integration for the uploaded files. This is because
+the NetBox charm is designed to work in a high availability (HA) configuration.
```

Comment:
Maybe add a note how S3 enables HA, I would assume it is about storing things only.

Reply:
Added extra info about the reason for S3 (basically upload images to S3 instead to the local filesystem).